realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -172,6 +172,11 @@ static synchronized void release(BaseRealm realm) {
             return;
         }
 
+        realm.tidyPerformedTransactions();
+        if (!realm.asyncTransactions.isEmpty() || !realm.asyncTransactionCallbacks.isEmpty()) {
```

Comment:
1. I am not quite sure if it worth to add a list `asyncTransactions` just for printing some warnings
2. If it does, better to have this logic in `BaseRealm.doClose()`
@realm/java